c7n/resources/cw.py

```diff
@@ -89,6 +90,36 @@ def process(self, resources):
                 AlarmNames=[r['AlarmName'] for r in resource_set])
 
 
+@Alarm.filter_registry.register('is-composite-child')
+class IsCompositeChild(Filter):
+    schema = type_schema('is-composite-child', state={"type": "boolean"})
+    permissions = ('cloudwatch:DescribeAlarms',)
+
+    def process(self, resources, event=None):
+        state = self.data.get("state", True)
+        # Get the composite alarms since filtered out in enum_spec
+        composite_alarms = self.manager.get_resource_manager("composite-alarm").resources()
+        composite_alarm_rules = jmespath_search('[].AlarmRule', composite_alarms)
+
+        parent_alarm_names = set()
+        # Loop through, find child alarm names
+        for rule in composite_alarm_rules:
+            names = self.extract_alarm_names_from_rule(rule)
+            parent_alarm_names.update(names)
+
+        if state:
+            # If we want to filter out alarms that are a child of a composite alarm
+            return [r for r in resources if r['AlarmName'] not in parent_alarm_names]
```

Comment:
Shouldn't this be the other way around? Meaning, if state is True, return resources not in `parent_alarm_names`?

Reply:
I also think that the name here doesn't suit well. `parent_alarm_names` could potentially be `child_alarm_names`?

Reply:
Good catch, updated.